app/lib/scorecard/backend.dart

```diff
@@ -74,33 +72,26 @@ class ScoreCardBackend {
       }
     }
 
+    if (onlyCurrent) return null;
+
     // List cards that at minimum have a pana report.
-    final query = _db.query<ScoreCard>(ancestorKey: key.parent);
-    final cardsWithPana = await query
-        .run()
-        .where((sc) =>
-            sc.runtimeVersion == versions.runtimeVersion ||
-            isNewer(sc.semanticRuntimeVersion, versions.semanticRuntimeVersion))
-        .where((sc) =>
-            !versions.blacklistedRuntimeVersions.contains(sc.runtimeVersion))
-        .where((sc) => sc.toData().hasReports([ReportType.pana]))
-        .toList();
-    final cardsWithAll = cardsWithPana
-        .where((sc) => sc.toData().hasReports(requiredReportTypes))
+    final fallbackKeys = versions.fallbackRuntimeVersions
+        .map(
+            (v) => scoreCardKey(packageName, packageVersion, runtimeVersion: v))
         .toList();
-
-    final cards = cardsWithAll.isNotEmpty ? cardsWithAll : cardsWithPana;
-    if (cards.isEmpty) {
-      return null;
-    }
-
-    final latest = cards.fold<ScoreCard>(
-        cards.first,
-        (latest, current) => isNewer(
-                latest.semanticRuntimeVersion, current.semanticRuntimeVersion)
-            ? current
-            : latest);
-    return latest.toData();
+    final fallbackCards = await _db.lookup<ScoreCard>(fallbackKeys);
```

Comment:
You so now you lookup all the fallback keys concurrently, I would suggest only looking up 5 keys at the time.. or maybe even 3..

If `fallbackRuntimeVersions` one day contains 50 entries, because we just didn't cleanup that seems problematic.. We can also fix this with a test that complains if `fallbackRuntimeVersions` has more than `N` entries :D

Reply:
Why would have 50? Those versions are unlikely to be running, and we don't want their data to be around...

Reply:
good point,